templates/common/.config/webpack/webpack.config.ts

```diff
@@ -12,10 +12,10 @@ import path from 'path';
 import ReplaceInFileWebpackPlugin from 'replace-in-file-webpack-plugin';
 import { Configuration } from 'webpack';
 
-import { getPackageJson, getPluginId, hasReadme } from './utils';
-import { SOURCE_DIR, DIST_DIR, ENTRY_FILE } from './constants';
+import { getPackageJson, getPluginId, hasReadme, getEntries } from './utils';
+import { SOURCE_DIR, DIST_DIR } from './constants';
 
-const config = (env): Configuration => ({
+const config = async (env): Promise<Configuration> => ({
```

Comment:
If we choose to do this we will need to update the docs as well to include how to use webpack-merge with a promise based webpack config.

Reply:
So the question is, should I rewrite this so it is using the sync functions? Then we don't need to change so the configuration function returns a Promise.

Reply:
On the other hand. We probably need to do this sooner or later 🙈 

What docs do we have that uses the `webpack-merge`?

Reply:
The [readme](https://github.com/grafana/create-plugin/blob/main/templates/common/.config/README.md) contains an example of how to extend this webpack config. I say we go with this change now as doing it later is only going to be more painful.